GoogleForCreators/web-stories-wp
# Dashboard: Replace Template Details with Images

## Context

Part of efforts to remove dependencies on the editor in the dashboard. 

## Summary

Instead of rendering each template as a story in the detail view, use screenshots. 

Perf metrics (macbook pro, high speed internet, chrome v 92.0.4515.159):
First load of a template details page (refresh and load dashboard with detail view): ~1493ms now vs. ~7478ms
Next template (arrow from detail view): ~100ms to load now vs ~1572ms

![Screen Shot 2021-08-26 at 3 25 02 PM](https://user-images.githubusercontent.com/10720454/131044092-ce97e4b4-f803-4594-a6a2-d5f9f40eaae9.png)
![demo detail page](https://user-images.githubusercontent.com/10720454/131044107-1d0102ca-993c-4775-a041-9893ce18f0e3.gif)

## Relevant Technical Choices

- Cleaned up the `cardGallery` (not convinced this belongs in `/components`, think it should be in the `detailTemplate` view since it's specific - left it in place for easier diff reading). There's no need for recalculating the size anymore because we can respect the ratio of the image and just set up a grid instead. 
-  Added the poster images to templates when we grab them in `useTemplates`, updated explore templates and karma data accordingly. 

## To-do

- [x] Grab screenshots 
- [x] Grab perf metrics 
- [x] Double check for any lingering `FontProviders`, `TransformProviders`, etc.  🔥 
- [x] Testing directions

This clears the path for #8746 (that'll remove the final dependencies on the story-editor package) as well as a deeper perf dive for the dashboard now that we can see what else is shifting the page layout or rendering maybe a few more times than should be necessary. 

## User-facing changes

- Template detail grid now uses screenshots instead of rendering template as a story (no animations) 
- There should be a purple border around the selected corresponding small grid item to the larger (selected) page. This matches #8783 update for links on my stories/dashboard. 
- Mostly it should just load faster. 

## Testing Instructions

- Load the dashboard, go to explore templates. Verify explore templates is still loading the first page of the template as a screenshot (did some behind the scenes work, so regression testing on that is good). 
- Click on 'see details' of any template, page should load as usual. Verify that template details view has screenshots now. 
- Verify that gallery of template details pages is navigable by keyboard and with a cursor. Clicking any of the pages will load it in the larger view. 
- You can create a story from a template in the template detail view.




This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

Cleaning up the detail template view, realized we're tracking template usage from the detail view but not the gallery - updated so we're tracking the same info from the gallery as we are the detail view. 

### Does this PR have a legal-related impact?

No

## Checklist




---



Partially addresses #8373
